corehq/apps/app_manager/templates/app_manager/partials/build_errors.html

```diff
@@ -317,5 +317,12 @@
                 {% trans "Errors Building an Application" %}
             </a>
         </p>
+        {% if in_iframe %}
+        <script>
+            $('#build-errors a').each(function(index) {
+                $(this).attr('target', '_parent');
+            });
+        </script>
+        {% endif %}
```

Comment:
the example of `app_manager.js` wouldn't work because the partial rendered in preview app doesn't have that script. the partial in general is generated like this: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/views/cli.py#L97

and then we return that html in a json call from formplayer. it's arguable a little less clear if you need to have another script loaded in order for that page to render correctly.

the example js code you gave i don't think would work since i _think_ document would refer to the iframe document and thus change the URL of the iframe, not the parent window. though definitely plausible that with a little hacking we could make that work. happy to quickly sync offline